pkg/graveler/ref/manager.go

```diff
@@ -561,3 +565,38 @@ func newCache(cfg *CacheConfig, def *CacheConfig) cache.Cache {
 	}
 	return c
 }
+
+func (m *Manager) SetAddressToken(ctx context.Context, repository *graveler.RepositoryRecord, token string) error {
+	a := &graveler.AddressData{
+		Address:   token,
+		ExpiredAt: timestamppb.New(time.Now().Add(AddressTokenTime)),
+	}
+	err := m.kvStore.SetMsgIf(ctx, graveler.RepoPartition(repository), []byte(graveler.AddressPath(token)), a, nil)
+	if err != nil {
+		if errors.Is(err, kv.ErrPredicateFailed) {
+			err = graveler.ErrAddressTokenAlreadyExists
+		}
+		return err
+	}
+	return nil
+}
+
+func (m *Manager) GetAddressToken(ctx context.Context, repository *graveler.RepositoryRecord, token string) error {
+	data := graveler.AddressData{}
+	path := []byte(graveler.AddressPath(token))
+	_, err := m.kvStore.GetMsg(ctx, graveler.RepoPartition(repository), path, &data)
+	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			err = graveler.ErrAddressTokenNotFound
+		}
+		return err
+	}
+	if data.ExpiredAt.AsTime().Before(time.Now()) {
+		return graveler.ErrAddressTokenExpired
+	}
+	return m.kvStore.DeleteMsg(ctx, graveler.RepoPartition(repository), path)
```

Comment:
I think it's best to separate the Get and Delete functionalities. It can cause confusion and it doesn't really save us any operations

Reply:
done

Reply:
I meant that we also should not delete the address as part of the GetAddressToken functionality but call it explicitly where needed